### PR TITLE
Engine: fill dummy textures with zeroes.

### PR DESCRIPTION
**Pull request by prideout**
We should to initialize these textures to prevent warnings with Vulkan.

### CODE REVIEW

**Review by bejado — APPROVED** [review]
LGTM, but maybe we only need to do this in debug builds? We shouldn't be sampling from these textures I don't think.

**Comment by prideout** [comment]
Interesting idea, we could skip this in release builds but then I would need to add some logic to the Vulkan backend to ensure that the image layout gets transitioned, since it wouldn't be spec compliant otherwise.

**Comment by bejado** [comment]
> Interesting idea, we could skip this in release builds but then I would need to add some logic to the Vulkan backend to ensure that the image layout gets transitioned, since it wouldn't be spec compliant otherwise.

I see, probably not worth the extra complexity then since these are super small textures.

**Review by pixelflinger — CHANGES_REQUESTED** [review]
WTH, we were not doing that? What was I drinking.

Actually I think some of those textures are actually read. That's why they're called DummyOne or DummyZero.

So the content needs to be correct, i.e. initialized with 0 or 1 (0xFF for RGBA8).

**Comment by prideout** [comment]
Doh, I updated the commit to fill the "ones" ones with ones.  :)

One interesting detail is that the tangents are **signed** integers, so I filled those with 0x7f...

File: filament/src/Engine.cpp [hunk 1]
```diff
@@ -288,12 +282,35 @@ void FEngine::init() {
...
             TextureFormat::RGBA8I, 1, 1, 1, 1, TextureUsage::DEFAULT);
 
     mDummyZeroTexture = driverApi.createTexture(SamplerType::SAMPLER_2D, 1,
             TextureFormat::RGBA8, 1, 1, 1, 1, TextureUsage::DEFAULT);
 
+    // initialize the dummy textures so that their contents are not undefined
+
+    using PixelBufferDescriptor = Texture::PixelBufferDescriptor;
+
+    static uint32_t zeroes = 0;
```
Comment (pixelflinger) [inline filament/src/Engine.cpp:295]: Use static const.

**Review by pixelflinger — CHANGES_REQUESTED** [review]
Use static const for the pixel variable.

File: filament/src/Engine.cpp (hunk 1 above)
Comment (prideout) [reply filament/src/Engine.cpp]: Done.

**Pull request merged**